### ISSUE TITLE
File browser opens to 'Recent' directory instead of the default download location when downloading blobs/files on Linux

### ISSUE THREAD

**Issue by v-kellyluo** [microsoft/AzureStorageExplorer#8232]
**Storage Explorer Version:** 1.36.0 (101)
**Build Number:** 20241012.1
**Branch:** rel/1.36.0
**Platform/OS:** Linux Ubuntu 22.04
**Architecture:** x64
**How Found:** From running test cases
**Regression From:** Previous release (1.34.0)

## Steps to Reproduce ##

1. Open 'Settings -> Transfers -> Default Download Location' -> Type a valid path in the textbox.
2. Open one blob container -> Select a blob -> Click 'Download'.
3. Check whether file browser opens to the default download location.

## Expected Experience ##
File browser opens to the default download location.

## Actual Experience ##
File browser opens to 'Recent' location.

## Additional Context ##

1. This issue doesn't reproduce for tables/disks.
2. This issue doesn't reproduce on Windows x64/Windows arm64/Mac arm64.

**Comment by v-kellyluo** [comment]
Hi @craxal , this issue reproduces for snaps and .tar.gz, and it doesn't reproduce on 1.34.0, here are the records:
1. On 1.34.0:
![Image](https://github.com/user-attachments/assets/cf1292c3-abde-4176-b2b4-9d7922bce687)
2. On build 20241015.1:
![Image](https://github.com/user-attachments/assets/acc223da-ae19-44cd-8f5b-1fedfc88bbe7)
3. In addition, I am not sure whether the bug you are talking about is [5363](https://github.com/microsoft/AzureStorageExplorer/issues/5363), but this bug is different from it. For this bug, we need to set a default download location first, and check if the file browser opens to the default location.

**Comment by craxal** [comment]
@v-kellyluo Yes, I think that's the one. The bug may be different, but I believe the root cause is the same, because they all eventually invoke a native file dialog. And the problem with the native dialog has been known for a long time.

As long as the desired location can still be navigated, I do not think this is a blocking issue. We can look into it more after 1.36.0.

**Comment by v-kellyluo** [comment]
@craxal Yes, we can manually select the destination location, will verify this bug after it is fixed.

**Comment by craxal** [comment]
Found a possible solution/workaround: https://snapcraft.io/docs/xdg-desktop-portals.

In a nutshell, we can drop the `home` and `removable-media` interfaces and add the `desktop` interface. Since Electron already supports desktop portals, that should theoretically resolve this issue.

**Comment by craxal** [comment]
Since this is not blocking anyone, we won't focus on looking for a fix.
